src/nbl/video/COpenGLCommandBuffer.cpp

```diff
@@ -503,6 +503,28 @@ COpenGLCommandBuffer::~COpenGLCommandBuffer()
 
     void COpenGLCommandBuffer::executeAll(IOpenGL_FunctionTable* gl, SOpenGLContextLocalCache* ctxlocal, uint32_t ctxid) const
     {
+        IGPUCommandPool::CommandSegment::Iterator itr = m_segmentListHeadItr;
+
+        if (itr.m_segment && itr.m_cmd)
+        {
+            while (itr.m_cmd->m_size != 0u)
+            {
+                COpenGLCommandPool::ICommand* glcmd = dynamic_cast<COpenGLCommandPool::ICommand*>(itr.m_cmd);
+                glcmd->operator()(gl, ctxlocal, ctxid, m_logger.getOptRawPtr(), this);
+
+                itr.m_cmd = reinterpret_cast<IGPUCommandPool::ICommand*>(reinterpret_cast<uint8_t*>(itr.m_cmd) + itr.m_cmd->m_size);
+                if ((reinterpret_cast<uint8_t*>(itr.m_cmd) - itr.m_segment->m_data) > IGPUCommandPool::CommandSegment::STORAGE_SIZE)
+                {
+                    IGPUCommandPool::CommandSegment* nextSegment = itr.m_segment->params.m_next;
+                    if (!nextSegment)
+                        break;
+
+                    itr.m_segment = nextSegment;
+                    itr.m_cmd = reinterpret_cast<IGPUCommandPool::ICommand*>(itr.m_segment->m_data);
+                }
+            }
+        }
+#if 0
```

Comment:
if you accept my QoL idea, then we're going to be iterating over a different segmented tape here.

Reply:
Resolved.